src/compiler/resolutionCache.ts

```diff
@@ -14,6 +14,9 @@ namespace ts {
         invalidateResolutionOfFile(filePath: Path): void;
         createHasInvalidatedResolution(): HasInvalidatedResolution;
 
+        startCachingPerDirectoryResolution(): void;

```

Comment:
I would prefer a singe method taking a callback, using `try-finally` to ensure that needed cleanup happens.